pkg/kv/kvserver/replica_init.go

```diff
@@ -243,6 +253,25 @@ func (r *Replica) loadRaftMuLockedReplicaMuLocked(desc *roachpb.RangeDescriptor)
 	return nil
 }
 
+func (r *Replica) loadUninit(ctx context.Context, desc *roachpb.RangeDescriptor) error {
+	// FIXME: Leaving any of the following fields nil leads to panics in some
+	// tests. The safest would be to initialize ReplicaState to a reasonable
+	// default (same as stateLoader.Load on an empty storage would return). In
+	// contrast, revisiting nil checking everywhere seems too much work.
+	r.mu.state.Lease = &roachpb.Lease{}
+	r.mu.state.TruncatedState = &roachpb.RaftTruncatedState{}
+	r.mu.state.GCThreshold = &hlc.Timestamp{}
+	r.mu.state.Stats = &enginepb.MVCCStats{}
+	// FIXME: Only here to satisfy the assert below.
+	r.mu.state.GCHint = &roachpb.GCHint{}
+	// FIXME: r.mu.state.Version?
+
+	// FIXME: is this needed?
```

Comment:
You could extract a method `assertStateUninitedReplicaRaftMuLockedReplicaMuRLocked` out of  `assertStateRaftMuLockedReplicaMuRLocked` (and call it from there) and then invoke that new first method directly here. I guess it would only check the `RaftReplicaID` and zero `HardState.Commit` etc but maybe over time it would learn some more tricks, not sure.

Reply:
Leaving the whole assert in place for now.